src/liboslexec/batched_llvm_gen.cpp

```diff
@@ -4313,19 +4313,15 @@ LLVMGEN (llvm_gen_raytype)
     llvm::Value * sg = rop.sg_void_ptr();
     if (Name.is_constant()) {
         // We can statically determine the bit pattern
-        ustring name = ((ustring *)Name.data())[0];
-        llvm::Value *args[] = {
-            sg,
-            rop.ll.constant (rop.shadingsys().raytype_bit (name))};
-
-        llvm::Value *ret = rop.ll.call_function (rop.build_name("raytype_bit"), args);
-
-        if(!result_is_uniform)
-        {
+        llvm::Value* bit;
+        bit = rop.ll.constant(rop.shadingsys().raytype_bit(Name.get_string()));
+        llvm::Value* raytype = rop.ll.op_load_int(
```

Comment:
Wondering if less is more, meaning was op_load_int needed, in that LLVM_Util::ptr_to_cast or LLVM_Util::ptr_cast and LLVM_Util::op_load which could be used to provide the same functionality.